src/python/grpcio_tests/tests/_runner.py

```diff
@@ -194,24 +201,27 @@ def try_set_handler(name, handler):
                 sys.stdout.write('Running       {}\n'.format(
                     augmented_case.case.id()))
                 sys.stdout.flush()
-                case_thread = threading.Thread(
-                    target=augmented_case.case.run, args=(result,))
-                try:
-                    with stdout_pipe, stderr_pipe:
-                        case_thread.start()
-                        while case_thread.is_alive():
-                            check_kill_self()
-                            time.sleep(0)
-                        case_thread.join()
-                except:  # pylint: disable=try-except-raise
-                    # re-raise the exception after forcing the with-block to end
-                    raise
-                result.set_output(augmented_case.case, stdout_pipe.output(),
-                                  stderr_pipe.output())
-                sys.stdout.write(result_out.getvalue())
-                sys.stdout.flush()
-                result_out.truncate(0)
-                check_kill_self()
+                if self._dedicated_threads:
+                    case_thread = threading.Thread(
+                        target=augmented_case.case.run, args=(result,))
+                    try:
+                        with stdout_pipe, stderr_pipe:
```

Comment:
Some comments or well-named helper functions would be helpful here, I think.

Reply:
Comments added.